### ISSUE TITLE
tidb panic when execute insert statements.

### ISSUE THREAD

**Issue by crazycs520** [pingcap/tidb#55453]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
create table t (id int key, k int,c int);
alter table t add index idx(k) global;
insert into t values (1,1,1); -- this will panic
```

```log
[2024/08/16 00:25:13.086 +08:00] [WARN] [session.go:2157] ["run statement failed"] [conn=2097154] [session_alias=] [schemaVersion=58] [error="runtime error: invalid memory address or nil pointer dereference"] [session="{
  \"currDBName\": \"test\",
  \"id\": 2097154,
  \"status\": 2,
  \"strictMode\": true,
  \"user\": {
    \"Username\": \"root\",
    \"Hostname\": \"127.0.0.1\",
    \"CurrentUser\": false,
    \"AuthUsername\": \"root\",
    \"AuthHostname\": \"%\",
    \"AuthPlugin\": \"mysql_native_password\"
  }
}"]
[2024/08/16 00:25:13.086 +08:00] [INFO] [conn.go:1178] ["command dispatched failed"] [conn=2097154] [session_alias=] [connInfo="id:2097154, addr:127.0.0.1:51202 status:10, collation:utf8mb4_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="insert into t values (1,1,1)"] [txn_mode=PESSIMISTIC] [timestamp=451867866060554240] [err="runtime error: invalid memory address or nil pointer dereference
github.com/pingcap/errors.AddStack
	/Users/cs/code/goread/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20240318064555-6bd07397691f/errors.go:178
github.com/pingcap/errors.Trace
	/Users/cs/code/goread/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20240318064555-6bd07397691f/juju_adaptor.go:15
github.com/pingcap/tidb/pkg/util.GetRecoverError
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/util/util.go:303
github.com/pingcap/tidb/pkg/executor/internal/exec.Next.func1
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/internal/exec/executor.go:435
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:914
runtime.panicmem
	/usr/local/go/src/runtime/panic.go:261
runtime.sigpanic
	/usr/local/go/src/runtime/signal_unix.go:861
github.com/pingcap/tidb/pkg/table/tables.(*index).GenIndexKey
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/table/tables/index.go:84
github.com/pingcap/tidb/pkg/table/tables.(*index).create
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/table/tables/index.go:185
github.com/pingcap/tidb/pkg/table/tables.(*TableCommon).addIndices
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/table/tables/tables.go:1041
github.com/pingcap/tidb/pkg/table/tables.(*TableCommon).addRecord
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/table/tables/tables.go:953
github.com/pingcap/tidb/pkg/table/tables.(*TableCommon).AddRecord
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/table/tables/tables.go:742
github.com/pingcap/tidb/pkg/executor.(*InsertValues).addRecordWithAutoIDHint
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/insert_common.go:1416
github.com/pingcap/tidb/pkg/executor.(*InsertExec).exec
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/insert.go:112
github.com/pingcap/tidb/pkg/executor.insertRows
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/insert_common.go:252
github.com/pingcap/tidb/pkg/executor.(*InsertExec).Next
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/insert.go:339
github.com/pingcap/tidb/pkg/executor/internal/exec.Next
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/internal/exec/executor.go:451
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).next
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/adapter.go:1252
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).handleNoDelayExecutor
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/adapter.go:1001
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).handleNoDelay
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/adapter.go:834
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).Exec
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/executor/adapter.go:598
github.com/pingcap/tidb/pkg/session.runStmt
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/session/session.go:2284
github.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/session/session.go:2145
github.com/pingcap/tidb/pkg/server.(*TiDBContext).ExecuteStmt
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/server/driver_tidb.go:291
github.com/pingcap/tidb/pkg/server.(*clientConn).handleStmt
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/server/conn.go:2051
github.com/pingcap/tidb/pkg/server.(*clientConn).handleQuery
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/server/conn.go:1805
github.com/pingcap/tidb/pkg/server.(*clientConn).dispatch
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/server/conn.go:1379
github.com/pingcap/tidb/pkg/server.(*clientConn).Run
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/server/conn.go:1141
github.com/pingcap/tidb/pkg/server.(*Server).onConn
	/Users/cs/code/goread/src/github.com/pingcap/tidb/pkg/server/server.go:741
runtime.goexit
	/usr/local/go/src/runtime/asm_arm64.s:1197"]
```

### 4. What is your TiDB version? (Required)

```sql
> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v8.3.0-alpha-169-g89bb81b3a7
Edition: Community
Git Commit Hash: 89bb81b3a773093099a0c65d188899e9e51b3c14
Git Branch: release-8.3
UTC Build Time: 2024-08-14 03:43:37
GoVersion: go1.23.0
Race Enabled: false
Check Table Before Drop: false
Store: tikv
```

**Comment by crazycs520** [comment]
I found this already fixed by #55440

**Comment by mjonss** [comment]
Closing, since similar as #55424, fixed by #55440, also merged into 8.3 in #55451.
